lib/install/management/delete.go

```diff
@@ -238,11 +238,37 @@ func (d *Dispatcher) DeleteVCHInstances(vmm *vm.VirtualMachine, conf *config.Vir
 	var err error
 	var children []*vm.VirtualMachine
 
-	// TODO: GET CHILDREN EITHER BY FOLDER OR RP BASED ON DEPLOYMENT
-	if children, err = d.parentResourcepool.GetChildrenVMs(d.op, d.session); err != nil {
+	parentFolder, err := vmm.Folder(d.op)
+	if err != nil {
 		return err
 	}
 
+	if parentFolder.Reference() == d.session.VMFolder.Reference() {
+		// use the resource pool to cut down on the number of candidates
+		d.op.Debugf("looking in the resource pool for delete targets")
+		if children, err = d.parentResourcepool.GetChildrenVMs(d.op, d.session); err != nil {
+			return err
+		}
+	} else {
+		// vch parent inventory folder exists, get the children from it.
+		folderChildren, err := parentFolder.Children(d.op)
+		if err != nil {
+			return err
+		}
+
+		for _, child := range folderChildren {
+			vmObj, ok := child.(*object.VirtualMachine)
+			if ok {
+				childVM := vm.NewVirtualMachine(d.op, d.session, vmObj.Reference())
+				cerr := childVM.EnableDestroy(d.op)
+				if cerr != nil {
+					d.op.Debugf("unable to enable the destroy task on container (%s): due to %s", childVM.InventoryPath, cerr)
```

Comment:
Minor: s/unable/Unable for consistent casing in vic-machine.log messages.

Also, something like `"Unable to enable destroy task on container (%s): %s"` is more succinct.